cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -61,36 +61,114 @@ const (
 	PoolPrefix PoolNamePrefix = "cstor-"
 )
 
+type ImportOptions struct {
+	// CachefileFlag option to use cachefile for import
+	CachefileFlag bool
+
+	// DevPath is directory where pool devices resides
+	DevPath string
+
+	// dontImport, being true, makes sure `zpool import` command is built
+	// without pool name.
+	// This way, we know the existence of pool without importing pool
+	dontImport bool
+}
+
 // RunnerVar the runner variable for executing binaries.
 var RunnerVar util.Runner
 
 // ImportPool imports cStor pool if already present.
-func ImportPool(cStorPool *apis.CStorPool, cachefileFlag bool) error {
-	importAttr := importPoolBuilder(cStorPool, cachefileFlag)
+func ImportPool(cStorPool *apis.CStorPool, importOptions *ImportOptions) (string, error) {
+	var importAttr []string
+
+	importAttr = importPoolBuilder(cStorPool, importOptions)
+
 	stdoutStderr, err := RunnerVar.RunCombinedOutput(zpool.PoolOperator, importAttr...)
 	if err != nil {
-		glog.Errorf("Unable to import pool: %v, %v", err.Error(), string(stdoutStderr))
-		return err
+		glog.Errorf("Unable to import pool: %v, %v devpath: %v cacheflag: %v", err.Error(),
+			string(stdoutStderr), importOptions.DevPath, importOptions.CachefileFlag)
+		return string(stdoutStderr), err
 	}
-	glog.Info("Importing Pool Successful")
-	return nil
+
+	glog.Infof("Import command successful with %v %v dontimport: %v", importOptions.CachefileFlag,
+		importOptions.DevPath, importOptions.dontImport)
+	return string(stdoutStderr), nil
 }
 
 // importPoolBuilder is to build pool import command.
-func importPoolBuilder(cStorPool *apis.CStorPool, cachefileFlag bool) []string {
+func importPoolBuilder(cStorPool *apis.CStorPool, importOptions *ImportOptions) []string {
 	// populate pool import attributes.
 	var importAttr []string
+	var cachefileFlag bool
+	var devPath string
+
+	// import takes either cachefile or devPath, not both.
+	// here, prioritizing devPath. Using zero values for cachefileFlag and devPath
+	if importOptions.DevPath != "" {
+		devPath = importOptions.DevPath
+	} else {
+		cachefileFlag = importOptions.CachefileFlag
+	}
+
 	importAttr = append(importAttr, "import")
 	if cStorPool.Spec.PoolSpec.CacheFile != "" && cachefileFlag {
-		importAttr = append(importAttr, "-c", cStorPool.Spec.PoolSpec.CacheFile,
-			"-o", cStorPool.Spec.PoolSpec.CacheFile)
+		importAttr = append(importAttr, "-c", cStorPool.Spec.PoolSpec.CacheFile)
+	}
+
+	importAttr = append(importAttr, "-o cachefile=", cStorPool.Spec.PoolSpec.CacheFile)
```

Comment:
Because of below snippet
```
	if cStorPoolGot.Spec.PoolSpec.CacheFile == "" {
		cStorPoolGot.Spec.PoolSpec.CacheFile = "/tmp/pool1.cache"
	}
```
every time we are importing by using cacheFile is it correct?

Reply:
'-o' is to set the cachefile name in which all the pool labels will be saved
'-c' is to set the cachefile name which can be used for import